gdk4/src/auto/texture_downloader.rs

```diff
@@ -0,0 +1,84 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::{MemoryFormat, Texture};
+use glib::{prelude::*, translate::*};
+use std::mem;
+
+glib::wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct TextureDownloader(Boxed<ffi::GdkTextureDownloader>);
+
+    match fn {
+        copy => |ptr| ffi::gdk_texture_downloader_copy(ptr),
+        free => |ptr| ffi::gdk_texture_downloader_free(ptr),
+        type_ => || ffi::gdk_texture_downloader_get_type(),
+    }
+}
+
+impl TextureDownloader {
+    #[doc(alias = "gdk_texture_downloader_new")]
+    pub fn new(texture: &impl IsA<Texture>) -> TextureDownloader {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gdk_texture_downloader_new(
+                texture.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_texture_downloader_download_bytes")]
+    pub fn download_bytes(&self) -> (glib::Bytes, usize) {
+        unsafe {
+            let mut out_stride = mem::MaybeUninit::uninit();
+            let ret = from_glib_full(ffi::gdk_texture_downloader_download_bytes(
+                self.to_glib_none().0,
+                out_stride.as_mut_ptr(),
+            ));
+            (ret, out_stride.assume_init())
+        }
+    }
+
+    //#[doc(alias = "gdk_texture_downloader_download_into")]
+    //pub fn download_into(&self, data: &[u8], stride: usize) {
```

Comment:
Adding this requires adding checks for the size of the slice and the validity of the stride